reference-implementation/lib/writable-stream.js

```diff
@@ -1029,13 +1026,15 @@ function defaultWriterReadyPromiseReset(writer) {
     writer._readyPromise_resolve = resolve;
     writer._readyPromise_reject = reject;
   });
+  writer._readyPromiseStatus = 'pending';
```

Comment:
If we're going to have a [[readyPromiseStatus]] slot, it should be explicitly initialised in the constructor.